content/blog/2020/01/2019-10-04-vscode-caseStudy.adoc

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+title: "Validating JCasC configuration files using Visual Studio Code"
+opengraph:
+  image: /images/logos/JCasC/JCasC.png
+tags:
+- community-bridge
+- JCasC
+- VSCode
+- community
+author: sladyn98
+---
+
+
+==== Configuration-as-code plugin
+The link:https://plugins.jenkins.io/configuration-as-code[Configuration as Code plugin] has been designed as an opinionated way to configure Jenkins based on human-readable declarative configuration files. Writing such a file should be feasible without being a Jenkins expert, just translating into code a configuration process one is used to executing in the web UI. The plugin uses a schema to verify the files being applied to the Jenkins instance.
+
+With the new JSON Schema being enabled developers can now test their yaml file against it. The schema checks the `descriptors` i.e. configuration that can be applied to a plugin or Jenkins core, the correct type is used and help text is provided in some cases. VSCode allows us to test out the schema right out of the box with some modifications. This project was built as part of the link:https://communitybridge.org/[Community Bridge] initiative which is a platform created by the Linux Foundation to empower developers — and the individuals and companies who support them — to advance sustainability, security, and diversity in open source technology. 
+
+=== Steps to Enable the Schema Validation
+
+a) The first step includes installing the link:https://marketplace.visualstudio.com/items?itemName=jcasc-developers.jcasc-plugin[JCasC-Plugin] and opening up the extension via the extention list. Shortcut for opening the extension list in VSCode editor using `Ctrl + Shift + X`.
+
+b) The next step would be  inserting the JSON Schema inside the VSCode workspace. We would be including the schema in the current directory.
+
+c) Spin up your Jenkins server and navigate to `/configuration-as-code/jsonSchema`. 
```

Comment:
aren't we using the plugin for this?

Reply:
Yes we are, I guess we added that there so that if the user wanted to manually add the schema on his own then he could do so, or else the extension just does the work for me.
Open to removing this and skipping to point D if you feel that is less confusing

Reply:
please do